lib/hsa/mcwamp_hsa.cpp

```diff
@@ -4454,30 +4454,61 @@ hsa_status_t HSACopy::hcc_memory_async_copy(Kalmar::hcCommandKind copyKind, cons
 
     hsa_agent_t hostAgent = const_cast<Kalmar::HSADevice *> (copyDeviceArg)->getHostAgent();
 
-    void *dstPtr = dstPtrInfo._devicePointer;
-    void *srcPtr = srcPtrInfo._devicePointer;
+    /* Determine src and dst pointer passed to ROCR runtime.
+     *
+     * Pre-condition:
+     * - this->dst and this->src must be tracked by AM API implemantation.
+     * - dstPtrInfo and srcPtrInfo must be valid.
+     */
+    void *dstPtr = nullptr;
+    void *srcPtr = nullptr;
 
     hsa_agent_t srcAgent, dstAgent;
     switch (copyKind) {
         case Kalmar::hcMemcpyHostToHost: 
             srcAgent=hostAgent; dstAgent=hostAgent;
 
-            // Use host pointers since this copy will be performed with CPU.
-            // If pointers are registered, then devicePointer may not match host pointer.
```

Comment:
@bensander I'd like to ask for your clarification on whether to keep the comment only, or should we continue to use `dstPtrInfo._hostPointer` in H2H case? My interpretation is the former one, not the later one. But I'd like to get your consent. Thanks.